js/content/community.js

```diff
@@ -571,17 +547,26 @@ let ProfileHomePageClass = (function(){
             htmlstr += CommunityCommon.makeProfileLink("custom", link, name, iconType, icon);
         }
 
-        // profile permalink
-        if (SyncedStorage.get("profile_permalink")) {
-            let imgUrl = ExtensionResources.getURL("img/clippy.svg");
-            htmlstr +=
-                `<div id="es_permalink_div" class="profile_count_link">
-					<span class="count_link_label">${Localization.str.permalink}</span>
-					<div class="es_copy_wrap">
-						<input id="es_permalink" type="text" value="https://steamcommunity.com/profiles/${steamId}" readonly />
-						<button id="es_permalink_copy"><img src="${imgUrl}" /></button>
-					</div>
-				</div>`;
+        // profile steamid
+        if (SyncedStorage.get("profile_steamid")) {
+            let dropdown = document.querySelector("#profile_action_dropdown .popup_body.popup_menu");
+            if (dropdown) {
+                dropdown.innerHTML += 
+                    `<a class="popup_menu_item" id="es_steamid">
+                        <img src="https://steamcommunity-a.akamaihd.net/public/images/skin_1/iconForums.png">&nbsp; ${Localization.str.view_steamid}
+                    </a>`;
+            } else {
+                let actions = document.querySelector(".profile_header_actions");
+                if (actions) {
+                    actions.innerHTML +=
+                        `<a class="btn_profile_action btn_medium" id="es_steamid">
+                            <span>${Localization.str.view_steamid}</span>
+                        </a>`;
+                }
+            }
+
+            document.querySelector("#es_steamid").addEventListener("click", showSteamIdDialog);
+            document.addEventListener("click", copySteamId);
```

Comment:
Can you register this in `showSteamIdDialog`?

Reply:
No, because this should only be attached once. If I put it in there, every time the dialog will be shown a new listener will be attached.

Reply:
It's more efficient if you unregister it when the dialog closes

Reply:
How so? How often are you gonna click on a page? Also, registering and unregistering to me seems more intensive.

Reply:
Implemented by using the `Messenger`.